### PR TITLE
feat(api-reference): the button ⚡

### PR DESCRIPTION
**Pull request by cameronrohani**
image
image

## Release Notes

- **New Features**
  - Introduced a sidebar button for improved navigation.
  - Added a `WatchModeToggle` component to enhance user control over watch functionality.
  - Integrated an `OpenApiClientButton` for direct access to OpenAPI documentation.

- **Enhancements**
  - Improved drag-and-drop functionality with a more responsive design.
  - Updated `ImportCollectionModal` for better interactivity and visual feedback.
  - Enhanced `DarkModeToggle` with a modern slider design.
  - Added a new `error` variant to the `ScalarModal` for improved visual feedback.

- **Bug Fixes**
  - Refined error handling and response management in URL fetching functions.

- **Documentation**
  - Updated comments and documentation for clarity on future enhancements.

### CODE REVIEW

**Review by kelseyll — APPROVED** [review]

**Comment by cameronrohani** [comment]
Scalar API client

File: packages/api-client/src/layouts/Web/ApiClientWeb.vue
Comment (hwkr) [inline packages/api-client/src/layouts/Web/ApiClientWeb.vue:1]: Does this need to be turned on in `ApiClientApp` as well?

File: packages/components/src/components/ScalarTooltip/ScalarTooltip.vue [hunk 1]
```diff
@@ -45,14 +47,16 @@ defineEmits<{
...
       <TooltipPortal to=".scalar-client">
-        <TooltipContent
-          :align="props.align"
-          class="scalar-app"
-          :class="props.class"
-          :side="props.side"
-          :sideOffset="props.sideOffset">
-          <slot name="content" />
-        </TooltipContent>
+        <template v-if="!props.disabled">
```
Comment (hwkr) [inline packages/components/src/components/ScalarTooltip/ScalarTooltip.vue:50]: Does the v-if not work on `<TooltipContent>`?

File: packages/api-client/src/components/ImportCollection/ImportCollectionModal.vue [hunk 2]
```diff
@@ -126,34 +162,49 @@ watch(
...
+            </div>
           </div>
-        </div>
+        </template>
+        <!-- Watch Mode -->
+        <template v-if="prefetchResult?.url">
+          <WatchModeToggle
+            v-model="watchMode"
+            :disableToolTip="true" />
+          <div>hi hello</div>
```
Comment (hwkr) [inline packages/api-client/src/components/ImportCollection/ImportCollectionModal.vue:183]: Should probably remove this before merging

**Review by hwkr — CHANGES_REQUESTED** [review]
Looks good! Main thing is remove the "hi hello" div 😂 

Also I feel like the import form could be centered to the page instead of between the bottom text and the top.

Google Chrome-2024-10-22-17-32-13@2x

File: packages/api-client/src/components/ImportCollection/ImportCollectionModal.vue (hunk 2 above)
Comment (cameronrohani) [reply packages/api-client/src/components/ImportCollection/ImportCollectionModal.vue:183]: this is a WIP will push something up in next hour or two :)

File: packages/api-client/src/components/ImportCollection/ImportCollectionModal.vue [hunk 3]
```diff
@@ -87,19 +87,41 @@ const toggleBodyClass = (add: boolean) => {
   }
 }
 
-// Watch for changes in the URL or modal state
-watch([hasUrl, () => modalState.open], ([newHasUrl, newIsVisible]) => {
-  toggleBodyClass(!!newHasUrl && newIsVisible)
-})
+// Add this new function
```
Comment (hanspagel) [inline packages/api-client/src/components/ImportCollection/ImportCollectionModal.vue:90]: 🤔

File: packages/api-client/src/layouts/Web/ApiClientWeb.vue
Comment (hanspagel) [reply packages/api-client/src/layouts/Web/ApiClientWeb.vue:1]: Not yet! That’ll be a (slightly?) different flow.

**Review by marclave — APPROVED** [review]

**Pull request merged**